### PR TITLE
Dev/34 Remove students in group And Group Index Page

### PR DESCRIPTION
**Pull request by Chaoyuuu**
image

image

### CODE REVIEW

File: src/app/commons/Spinner.scss [hunk 1]
```diff
@@ -0,0 +1,88 @@
+.sk-folding-cube {
```
Comment (Johnny850807) [inline src/app/commons/Spinner.scss:1]: this should be a `css` file since

File: src/app/commons/modals/RemoveConfirmationModal.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
         show, onClose, closeIconRef,
         contentRendering: () => (
             <form onSubmit={handleSubmit} ref={formRef}>
                 <div id="modal" className="font-poppins has-text-centered">
                     <ModalHeader className="header" title={title} textAlign="center"/>
-                    <p>Name: {content.name}</p>
-                    <p>Email: {content.email}</p>
+                    {
+                        content?.map(item =>
+                            <p>{item.title}: {item.value}</p>
```
Comment (Johnny850807) [inline src/app/commons/modals/RemoveConfirmationModal.js:23]: You must add a `key` to item to distinguish them 
read [this](https://reactjs.org/docs/lists-and-keys.html)

`<p key={item.title}>{item.title}: {item.value}</p>`

File: src/app/commons/IndexBanner/IndexBanner.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import FakeLink from "../FakeLink";
+
+function link(currentPathName, to, name, Icon) {
+    return (
+        <NavLink to={to} activeClassName="is-active">
+            <Icon className="tab-icon"/>{name}</NavLink>
+    );
+}
+
+const IndexBanner = function ({currentPathName, path, tabContents}) {
```
Comment (Johnny850807) [inline src/app/commons/IndexBanner/IndexBanner.js:12]: 1. This is not a Banner
this is a navigation bar (or any better ideas?).

Rename it to `InPageNavigationBar` (remember to rename all its related files e.g., css)

2. A comment that documents how to use this component is required

File: src/app/commons/IndexBanner/IndexBanner.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    return (
+        <div className="index-container">
```
Comment (Johnny850807) [inline src/app/commons/IndexBanner/IndexBanner.js:14]: rename `index-container` to `in-page-nav-bar`

File: src/app/commons/modals/RemoveConfirmationModal.js [hunk 5]
```diff
@@ -1,25 +1,28 @@
-import './RemoveExamineeConfirmationModal.scss'
+import './RemoveConfirmationModal.scss'
 import React, {createRef} from "react";
-import {renderModal} from "../../commons/modals/modal";
-import {ModalHeader} from "../../commons/modals/ModalHeader";
+import {renderModal} from "./modal";
+import {ModalHeader} from "./ModalHeader";
 
-const RemoveExamineeConfirmationModal = ({title, content, show, onClose}) => {
+const RemoveConfirmationModal = ({title, content, show, onClose}) => {
```
Comment (Johnny850807) [inline src/app/commons/modals/RemoveConfirmationModal.js:6]: 1. rename `content` to `data`
2. comment guide is required

File: src/app/students/GroupIndexBanner.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import {IndexBanner} from "../commons/IndexBanner/IndexBanner";
+import {AiOutlineSetting, FaUserGraduate} from "react-icons/all";
+
+const GroupIndexBanner = function ({currentPathName, groupName, groupId}) {
```
Comment (Johnny850807) [inline src/app/students/GroupIndexBanner.js:4]: rename

File: src/app/students/GroupMembers.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            studentService.getStudentsByGroupId(group.id)
+                .then(students => setStudents(students));
+        }
+    });
+
+    return (
+        <div>
+            {group === undefined ? <Spinner/> : <GroupIndexBanner currentPathName={currentPathName}
+                                                                  groupName={group.name}
+                                                                  groupId={group.id}/>}
```
Comment (Johnny850807) [inline src/app/students/GroupMembers.js:47]:
```js
 {group ? <GroupIndexBanner currentPathName={currentPathName} 
                                  groupName={group.name} groupId={group.id}/> : <Spinner/> }
```

( `GroupIndexBanner` should be renamed )

File: src/services/StudentService.js [hunk 8]
```diff
@@ -97,6 +97,16 @@ export default class StudentService {
             .then(res => res.data.map(obj => new Group(obj)));
     }
 
+    async getGroupById(groupId) {
+        return this.axios.get(`/api/groups/${groupId}`)
+            .then(res => new Group(res.data));
+    }
+
+    async getStudentsByGroupId(groupId) {
```
Comment (Johnny850807) [inline src/services/StudentService.js:105]: rename `getMembersInGroup`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/commons/Spinner.scss (hunk 1 above)
Comment (Chaoyuuu) [reply src/app/commons/Spinner.scss:1]: I changed it to `scss` style when you are reviewing. 😄

File: src/app/students/GroupMembers.js (hunk 7 above)
Comment (Johnny850807) [reply src/app/students/GroupMembers.js]: @Chaoyuuu Unsolved

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/students/GroupMembers.js (hunk 7 above)
Comment (Chaoyuuu) [reply src/app/students/GroupMembers.js]: image

I have renamed `GroupIndexBanner` to `GroupNavigationBar` in this pr.

Comment (Johnny850807) [reply src/app/students/GroupMembers.js]: no

**Review by Johnny850807 — APPROVED** [review]
ＬＧＴＭ

**Pull request merged**